### PR TITLE
Fixes N2O not decomposing above 100000 Kelvin

### PR DESCRIPTION
**Pull request by Migratingcocofruit**
## What Does This PR Do
What the title says

## Why It's Good For The Game
fixes a bug

## Testing
- filled a can with N2O
- set temp to 1401K and to 200000K
- Decomposition happened at the expected rate for each temperature

<hr>

### Declaration

- [x] I confirm that I either do not require [pre-approval](https://github.com/ParadiseSS13/Paradise/blob/master/docs/CODE_OF_CONDUCT.md#types-of-changes-that-need-approval) for this PR, or I have obtained such approval and have included a screenshot to demonstrate this below.
  
  
  <hr>

## Changelog

:cl:
fix: N2O now decomposes over 100000 Kelvin
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm [hunk 1]
```diff
@@ -574,21 +574,19 @@ What are the archived variables for?
 		var/energy_released = 0
 		var/old_heat_capacity = heat_capacity()
 		var/burned_fuel = 0
+		burned_fuel = (1 - (N2O_DECOMPOSITION_COEFFICIENT_A  / ((private_temperature + N2O_DECOMPOSITION_COEFFICIENT_C) ** 2))) * private_sleeping_agent
```
Comment (FunnyMan3595) [inline code/modules/atmospherics/gasmixtures/gas_mixture.dm:577]: Add a min() to this to be certain that it won't consume more sleeping agent than exists.

**Comment by silverplatedelta** [comment]
I will admit I do have come concerns about unchecked heat addition. Given that roughly 300,000 moles of N2 on round start, if done properly an atmos tech could easily heat gas to hundreds of thousands of degrees. I cant think of any real practical uses of doing this, I do want to point out that it can easily be done. 

I did some local testing and wasn't able to (easily) cause any serious problems with atmos.

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm (hunk 1 above)
Comment (Migratingcocofruit) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm:577]: The formula can only become greater than 1 or negative if the constants are changed. is this truly necessary?

**Comment by Migratingcocofruit** [comment]
> I will admit I do have come concerns about unchecked heat addition. Given that roughly 300,000 moles of N2 on round start, if done properly an atmos tech could easily heat gas to hundreds of thousands of degrees. I cant think of any real practical uses of doing this, I do want to point out that it can easily be done.
> 
> I did some local testing and wasn't able to (easily) cause any serious problems with atmos.

You could always get the same amount of heat out of it. This only changes the unexpected behaviour at 100000 kelvin

File: code/modules/atmospherics/gasmixtures/gas_mixture.dm (hunk 1 above)
Comment (FunnyMan3595) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm:577]: Unless we need the performance, it's better to make our guarantees explicit. We never want to burn more fuel than actually exists, so we should make it obvious that that's guaranteed.

Comment (Migratingcocofruit) [reply code/modules/atmospherics/gasmixtures/gas_mixture.dm]: done

**Pull request merged**